sktime/forecasting/trend/_stl_forecaster.py

```diff
@@ -327,6 +328,37 @@ def _update(self, y, X=None, update_params=True):
         self.forecaster_resid_.update(y=self.resid_, X=X, update_params=update_params)
         return self
 
+    def plot_components(self, title=None):
+        """Plot the observed, trend, seasonal, and residual components.
+            Requires state to be "fitted", i.e., ``self.is_fitted=True``.
+        """
+        _check_soft_dependencies("matplotlib")
```

Comment:
Passing `obj="STLForecaster.plot_components` will lead to a more informative error message.